sbol_utilities/sbol3_genbank_conversion.py

```diff
@@ -1,48 +1,112 @@
+                                "test/test_files", "BBa_J23101.gb")
+SAMPLE_SBOL3_FILE_1 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "BBa_J23101_from_genbank_to_sbol3_direct.nt")
+SAMPLE_GENBANK_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports.gb")
+SAMPLE_SBOL3_FILE_2 = os.path.join(os.path.abspath(os.path.curdir), "test/test_files",
+                                "iGEM_SBOL2_imports_from_genbank_to_sbol3_direct.nt")
+GB2SO_MAPPINGS = {}
+SO2GB_MAPPINGS = {}
+SO_SEQ_FEAT_ROLE_NS = "http://identifiers.org/so/"
+GB2SO_MAPPINGS_CSV = os.path.join(os.path.abspath(os.path.curdir), "gb2so.csv")
+SO2GB_MAPPINGS_CSV = os.path.join(os.path.abspath(os.path.curdir), "so2gb.csv")
+
+
+class GenBank_SBOL3_Converter:
+    def create_GB2SO_role_mappings(self, gb2so_csv: str = GB2SO_MAPPINGS_CSV, so2gb_csv: str = SO2GB_MAPPINGS_CSV):
+        """Reads 2 CSV Files containing mappings for converting between GenBank and SO ontologies roles
+        :param gb2so_csv: path to read genbank to so conversion csv file
+        :param so2gb_csv: path to read so to genbank conversion csv file
+        """
+        with open(gb2so_csv, mode='r') as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            for row in csv_reader:
+                GB2SO_MAPPINGS[row["GenBank_Ontology"]] = row["SO_Ontology"]
+        with open(so2gb_csv, mode='r') as csv_file:
+            csv_reader = csv.DictReader(csv_file)
+            for row in csv_reader:
+                SO2GB_MAPPINGS[row["SO_Ontology"]] = row["GenBank_Ontology"]
+
+
+    def convert_genbank_to_sbol3(self, gb_file: str, sbol3_file: str = "sbol3.out", namespace: str =
+                                TEST_NAMESPACE, write: bool = False) -> sbol3.Document:
+        """Convert a GenBank document on disk into an SBOL3 document
+        Specifically, the GenBank document is first imported to SBOL2, then converted from SBOL2 to SBOL3
+
+        :param gb_file: path to read GenBank file from
+        :param sbol3_file: path to write SBOL3 file to, if write set to true
+        :param namespace: URIs of Components will be set to {namespace}/{genbank_id},
+                          defaults to "https://test.sbol3.genbank/" 
+        :param write: writes the generated sbol3 document in SORTED_NTRIPLES
+                          format to provided sbol3_file path
+        :return: SBOL3 document containing converted materials
+        """
+        # create sbol3 document, and record parser handler for gb file
+        sbol3.set_namespace(namespace)
+        doc = sbol3.Document()
+        # access records by parsing gb file using SeqIO class
+        records = list(SeqIO.parse(gb_file, "genbank").records)
+        for record in records:
+            # NOTE: Currently we assume only linear or circular topology is possible
+            COMP_TYPES.append(sbol3.SO_LINEAR 
```

Comment:
Will `COMP_TYPES` keep growing and growing in size with each record processed since it is a global variable?